src/main/kotlin/net/atos/zac/smartdocuments/rest/RESTMappedSmartDocumentsTemplateGroup.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+package net.atos.zac.smartdocuments.rest
+
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+
+@NoArgConstructor
+@AllOpen
+data class RESTMappedSmartDocumentsTemplateGroup(
+    var id: String,
+    var name: String,
+    var groups: Set<RESTMappedSmartDocumentsTemplateGroup>?,
+    var templates: Set<RESTMappedSmartDocumentsTemplate>?,
+)
+
+fun Set<RESTMappedSmartDocumentsTemplateGroup>.toStringRepresentation(): Set<String> {
+    val result = mutableSetOf<String>()
+    this.forEach { result.addAll(convertTemplateGroupToStringRepresentation(it, null)) }
+    return result
+}
+
+private fun convertTemplateGroupToStringRepresentation(
+    group: RESTMappedSmartDocumentsTemplateGroup,
+    parent: String?
+): Set<String> =
+    arrayOf(parent, "group.${group.id}.${group.name}").filterNotNull().joinToString(".").let { groupString ->
+        mutableSetOf(groupString).apply {
```

Comment:
here you also don't need a mutableSet I'm pretty sure

Reply:
We need it to support `mapTo` in lambda